src/redux/sagas/firestore-saga.tsx

```diff
@@ -1,28 +1,82 @@
-import { takeEvery, call, put } from 'redux-saga/effects';
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+import { addHours, isBefore } from 'date-fns';
+// Redux
+import { RootReducerState } from 'reducers/rootReducer';
 import { ReduxFirestoreType as t } from 'redux/redux-types/firestore-type';
-import { IReduxActions } from 'redux/interfaces/IReduxActions';
-import { IReduxDataSentToFireStore } from 'redux/reducers/firestore-reducer';
+import { IReduxAction } from 'redux/interfaces/IReduxAction';
+import { IReduxFirestoreSyncedState } from 'redux/reducers/firestore-reducer';
+
+// Constants
+import { REMOTE_CONFIG } from 'constants/remoteConfigConstants';
+import { API_REQUEST_TIMEOUT } from 'constants/appConstants';
+
+// Services
+import { firebaseRemoteConfig } from 'services/firebase';
+
+// Utils
+import httpRequest from 'utils/httpRequest';
 import { logBreadcrumb } from 'utils/common';
+import { purgeSensitiveDataFromState, validateStateToSync } from 'utils/firestore-helper';
 
 // Watcher Saga
 export default function* firestoreSaga() {
-  yield takeEvery(t.SEND_DATA_TO_FIRESTORE, sendDataToFirestoreSaga);
+  yield takeEvery(t.FIRESTORE_SYNC_STATE, syncStateWithFirestoreSaga);
 }
 
-function* sendDataToFirestoreSaga(actions: IReduxActions) {
-  const payload: IReduxDataSentToFireStore = yield call(sendDataToFirestore);
-  yield put(payload);
+const requestConfig = {
+  timeout: API_REQUEST_TIMEOUT,
+  headers: {
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+  },
+};
+
+function* syncStateWithFirestoreSaga(actions: IReduxAction) {
+  const state: RootReducerState = yield select();
+
+  const lastSyncedUtc: string | null = state?.firestore?.lastSyncedUtc ?? null;
+
+  try {
+    const payload: IReduxFirestoreSyncedState = yield call(syncStateWithFirestore, state, lastSyncedUtc);
+    logBreadcrumb(TAG, 'Synced state with Firestore');
+    yield put(payload);
+  } catch (e) {
+    logBreadcrumb(TAG, 'Failed to sync state with Firestore');
+    logBreadcrumb(TAG, e.toString());
+    yield put<IReduxAction>({ type: t.FIRESTORE_SYNC_STATE_ERROR, error: e });
+  }
 }
-const sendDataToFirestore = async (): Promise<IReduxDataSentToFireStore> => {
-  logBreadcrumb(TAG, 'sending data to firestore saga');
 
-  // Firestore code goes here...
+const syncStateWithFirestore = async (
+  state: RootReducerState,
+  lastSyncedUtc?: string,
+): Promise<IReduxFirestoreSyncedState> => {
+  logBreadcrumb(TAG, 'Attempting to sync state with Firestore', state.firestore);
+
+  // Check if recently synced
+  if (lastSyncedUtc) {
+    const now = new Date();
+    const lastSynced = new Date(lastSyncedUtc);
+    if (isBefore(now, addHours(lastSynced, 1))) throw new Error('State has recently been synced');
+  }
+
```

Comment:
This is a nice feature but it got me thinking about the support team and the event that they may need to see this data sooner than later... let's chat about options 👍